g2p_programs/models/managers/entitlement_manager.py

```diff
@@ -224,6 +291,43 @@ def prepare_entitlements(self, cycle, beneficiaries):
                 }
             )
 
+    def cancel_entitlements(self, cycle):
+        """Cancel Entitlements
+        Default Entitlement Manager :meth:`cancel_entitlements`
+        Cancel entitlements in a cycle
+
+        :param cycle: A recordset of cycle
+        :return:
+        """
+        # Get the entitlements in cycle
+        entitlements_count = cycle.get_entitlements(
+            ["draft", "pending_validation"],
+            entitlement_model="g2p.entitlement",
+            count=True,
+        )
+        if entitlements_count < self.MIN_ROW_JOB_QUEUE:
+            self._cancel_entitlements(cycle)
+        else:
+            self._cancel_entitlements_async(cycle, entitlements_count)
+
+    def _cancel_entitlements(self, cycle, offset=0, limit=None):
```

Comment:
I think it should also disable entitlements that have been validated but not redeemed.

Reply:
Fixed